### PR TITLE
Momento.cli version 0.26.5

### PR DESCRIPTION
**Pull request by malandis**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/93218)

### CODE REVIEW

**Comment by malandis** [comment]
@microsoft-github-policy-service agree [company="momento"]

File: manifests/m/momento/cli/0.26.3/momento.cli.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: momento.cli
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.installer.yaml:4]:
```suggestion
PackageIdentifier: Momento.MomentoCLI
```
The convention for naming PackageIdentifier in WinGet is generally `Publisher.PackageName` (pascal case)

File: manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: momento.cli
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml:4]:
```suggestion
PackageIdentifier: Momento.MomentoCLI
```
The convention for naming PackageIdentifier in WinGet is generally `Publisher.PackageName` (pascal case)

File: manifests/m/momento/cli/0.26.3/momento.cli.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.2.0.schema.json
+
+PackageIdentifier: momento.cli
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.yaml:4]:
```suggestion
PackageIdentifier: Momento.MomentoCLI
```
The convention for naming PackageIdentifier in WinGet is generally `Publisher.PackageName` (pascal case)

File: manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: momento.cli
+PackageVersion: 0.26.3
+PackageLocale: en-US
+Publisher: momento
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml:7]:
```suggestion
Publisher: Momento
```
Must match Control Panel / Apps&Features.

File: manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: momento.cli
+PackageVersion: 0.26.3
+PackageLocale: en-US
+Publisher: momento
+PublisherUrl: https://www.gomomento.com/
+Author: momento
+PackageName: Momento CLI
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml:10]:
```suggestion
PackageName: Momento
```
Must match Control Panel / Apps&Features

File: manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: momento.cli
+PackageVersion: 0.26.3
+PackageLocale: en-US
+Publisher: momento
+PublisherUrl: https://www.gomomento.com/
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml:8]:
```suggestion
PublisherUrl: https://www.gomomento.com/
PublisherSupportUrl: https://www.gomomento.com/contact-us
PrivacyUrl: https://www.gomomento.com/privacy-policy
```
Additional metadata

File: manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+License: Apache-2.0 License
+LicenseUrl: https://github.com/momentohq/momento-cli/blob/main/LICENSE
+ShortDescription: Official CLI for Momento Serverless Cache
+Description: momento is Momento Serverless Cache on the command line. It brings account creation, maintenance, cache control and data commands to the terminal.
+Moniker: momento
+Tags:
+- cli
+- command-line
+- tool
+- utility
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.locale.en-US.yaml:21]:
```suggestion
- utility
ReleaseNotes: |-
  0.26.3 (2023-01-06)
  Chores

  - Updated Cargo version to v0.26.3 (4f44f94)
  - sign windows installer (#219) (d0113a1)
ReleaseNotesUrl: https://github.com/momentohq/momento-cli/releases/tag/v0.26.3
PurchaseUrl: https://www.gomomento.com/pricing
Documentations:
- DocumentLabel: Momento Documentation
  DocumentUrl: https://docs.momentohq.com/
```
Additional metadata

File: manifests/m/momento/cli/0.26.3/momento.cli.installer.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+PackageVersion: 0.26.3
+Platform:
+- Windows.Desktop
+InstallerType: wix
+InstallModes:
+- interactive
+- silent
+- silentWithProgress
+UpgradeBehavior: install
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.installer.yaml:13]:
```suggestion
UpgradeBehavior: install
AppsAndFeaturesEntries:
  - DisplayVersion: 0.26.3.0
```
As the package is marketed with `0.26.3` version but the actual DisplayVersion written to Registry is `0.26.3.0`, we're going to use AppsAndFeaturesEntries to provided 1:1 match with Registry.

File: manifests/m/momento/cli/0.26.3/momento.cli.installer.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+PackageIdentifier: momento.cli
+PackageVersion: 0.26.3
+Platform:
+- Windows.Desktop
+InstallerType: wix
+InstallModes:
+- interactive
+- silent
+- silentWithProgress
```
Comment (mdanish-kh) [inline manifests/m/momento/cli/0.26.3/momento.cli.installer.yaml:12]:
```suggestion
- silentWithProgress
ReleaseDate: 2023-01-06
```
Additional metadata

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Review by mdanish-kh — COMMENTED** [review]
After making the above changes, you would also have to modify the directory structure to match the new PackageIdentifier. `manifests/m/momento/cli/0.26.3` should be changed to `manifests/m/Momento/MomentoCLI/0.26.3` (the directory structure is case-sensitive). Consequently the file names would also require renaming to match the new PackageId. `momento.cli.installer.yaml` becomes `Momento.MomentoCLI.installer.yaml` and so on.

One quick way to do this would be here in GitHub Web by pressing the <kbd>.</kbd> (dot) key that will open this PR in VSCode Online. From the explorer tab on the left, find the required files and restructure the directory appropriately. After that you can stage and commit your changes from the Source Control tab on the left (the changes are automatically pushed).

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
Marking as `Needs-Author-Feedback` until remaining suggestions are addressed.

**Comment by malandis** [comment]
@mdanish-kh thank you for the detailed feedback and suggestions. I was unaware of the built in VSCode support by way of `.` -- my mind is blown!

Regarding your suggestions:
1. I have committed the changes you suggested about URLs for purchase, documentation, and privacy. Thank you for investigating this.
2. The dynamic metadata we will add in a future PR linked [here](https://github.com/momentohq/momento-cli/issues/225)
3. Regarding the installer version vs the metadata version: the build number should not have been in the installer. I have built a new installer without the build number, so this mismatch will not be there in the future. I will issue a commit that points to this new installer. That will take care of the directory structure.
4. Regarding the publisher name vs package name -- I am modelling this after the GitHub CLI, which I found [here](https://github.com/microsoft/winget-pkgs/blob/master/manifests/g/GitHub/cli/2.9.0/GitHub.cli.locale.en-US.yaml). This is how I'd like ours to look roughly. Theirs has:
- PackageIdentifier: GitHub.cli
- Publisher: GitHub, Inc.
- PackageName: GitHub CLI
- (in the app registry): GitHub CLI

I would like ours to parallel this:
- PackageIdentifier: Momento.cli
- Publisher: Momento
- PackageName: Momento CLI
- (in the app registry): Momento CLI

I can adjust the installer metadata appropriately. Will this be acceptable?

**Comment by mdanish-kh** [comment]
@malandis Sounds good! Thank you for taking the time to make changes to the installer.

> I would like ours to parallel this:
>  
> * PackageIdentifier: Momento.cli

Just a little note that you would still have to rename `momento` directory to `Momento` as PackageId and directory names are case-sensitive

**Comment by malandis** [comment]
> @malandis Sounds good! Thank you for taking the time to make changes to the installer.
> 
> > I would like ours to parallel this:
> > 
> > * PackageIdentifier: Momento.cli
> 
> Just a little note that you would still have to rename `momento` directory to `Momento` as PackageId and directory names are case-sensitive

Done! Committed in c067677

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

File: manifests/m/Momento/cli/0.26.5/Momento.cli.locale.en-US.yaml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+PublisherSupportUrl: https://www.gomomento.com/contact-us
+PrivacyUrl: https://www.gomomento.com/privacy-policy
+Author: Momento
+PackageName: Momento CLI
+PackageUrl: https://github.com/momentohq/momento-cli
+License: Apache-2.0 License
+LicenseUrl: https://github.com/momentohq/momento-cli/blob/main/LICENSE
+ShortDescription: Official CLI for Momento Serverless Cache
+Description: Momento CLI is Momento Serverless Cache on the command line. It brings account creation, maintenance, cache control and data commands to the terminal.
+Moniker: Momento
```
[the remark this answers is not in the document]
Comment (malandis) [reply manifests/m/Momento/cli/0.26.5/Momento.cli.locale.en-US.yaml]: Fixed in https://github.com/microsoft/winget-pkgs/pull/93218/commits/aa4e93f9d27bb5088da53f6f77df887aeb1c22b0

**Review by mdanish-kh — APPROVED** [review]

**Pull request merged**